File: app/src/main/java/com/example/bhagavadgita/Adapters/TranslationAdapter.java
package com.example.bhagavadgita.Adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.bhagavadgita.Models.Translation;
import com.example.bhagavadgita.R;

import java.util.ArrayList;

public class TranslationAdapter extends RecyclerView.Adapter<TranslationAdapter.ViewHolder> {

    ArrayList<Translation> translations;

    Context context;



    public TranslationAdapter(ArrayList<Translation> translations, Context context) {
        this.translations = translations;
        this.context = context;

    }



    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_single_verses, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {

            holder.tvDesc.setText(translations.get(position).getDescription());
            holder.tvLanguage.setText(translations.get(position).getLanguage().toUpperCase());
            holder.tvAuthor.setText(translations.get(position).getAuthor_name());

    }

    @Override
    public int getItemCount() {
            return translations.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        TextView tvDesc;
        TextView tvAuthor;
        TextView tvLanguage;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            tvDesc = itemView.findViewById(R.id.tvDesc);
            tvAuthor = itemView.findViewById(R.id.tvAuthor);
            tvLanguage = itemView.findViewById(R.id.tvLanguage);
        }
    }
}

File: app/src/main/java/com/example/bhagavadgita/ui/verses/adapter/TranslationAdapter.kt
package com.example.bhagavadgita.ui.verses.adapter

import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.bhagavadgita.data.model.Translation
import com.example.bhagavadgita.R
import com.google.gson.Gson
import java.util.Locale

class TranslationAdapter(var translations: ArrayList<Translation>, var context: Context) :
    RecyclerView.Adapter<TranslationAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.item_single_verses, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.tvDesc.text = translations[position].description
        holder.tvLanguage.text = translations[position].language!!.uppercase(Locale.getDefault())
        holder.tvAuthor.text = translations[position].author_name
    }

    override fun getItemCount(): Int {
        return translations.size
    }

    fun addData(translations: ArrayList<Translation>) {
        this.translations.addAll(translations)
    }
    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var tvDesc: TextView
        var tvAuthor: TextView
        var tvLanguage: TextView

        init {
            tvDesc = itemView.findViewById(R.id.tvDesc)
            tvAuthor = itemView.findViewById(R.id.tvAuthor)
            tvLanguage = itemView.findViewById(R.id.tvLanguage)
        }
    }
}
